src/python/pants/engine/internals/graph.py

```diff
@@ -101,8 +137,40 @@ async def resolve_target(
 
 @rule
 async def resolve_targets(addresses: Addresses) -> Targets:
+    # TODO: This method duplicates `resolve_targets_with_origins`, because direct expansion of
+    # `Addresses` to `Targets` is common in a few places: we can't always assume that we
+    # have `AddressesWithOrigins`. One way to dedupe these two methods would be to fake some
+    # origins, and then strip them afterward.
+    wrapped_targets = await MultiGet(Get(WrappedTarget, Address, a) for a in addresses)
+    targets = {wt.target for wt in wrapped_targets}
```

Comment:
Consider using `OrderedSet`. `set` is non-deterministic, and it looks like we don't sort this value ever. Alternatively, sort at line 167.